Tools/Anonymization/main.py

```diff
@@ -0,0 +1,140 @@
+    "ipv4": r'(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(?:\.)){3}(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?))',
+    "ipv6": r'(?:(?:[0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|(?:[0-9a-fA-F]{1,4}:){1,7}:|(?:[0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|(?:[0-9a-fA-F]{1,4}:){1,5}(?::[0-9a-fA-F]{1,4}){1,2}|(?:[0-9a-fA-F]{1,4}:){1,4}(?::[0-9a-fA-F]{1,4}){1,3}|(?:[0-9a-fA-F]{1,4}:){1,3}(?::[0-9a-fA-F]{1,4}){1,4}|(?:[0-9a-fA-F]{1,4}:){1,2}(?::[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:(?:(?::[0-9a-fA-F]{1,4}){1,6})|:(?:(?::[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(?::[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(?:ffff(?::0{1,4}){0,1}:){0,1}(?:(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])|(?:[0-9a-fA-F]{1,4}:){1,4}:(?:(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9]))',  
+    "name": r'((?<=name).+)',
+    "author": r'(?<=author).+',
+    "id": r'[S|T|F|G|A]\d{7}[A-Z]',
+    "class": r'(?<=class).+',
+    "group": r'(?<=group).+',
+    "section": r'(?<=section).+',
+    "tutorial": r'(?<=tutorial).+',
+    "github_id": r'@([a-z0-9](?:-?[a-z0-9]){0,38})',
+
+}
+
+class Anonymizer():
+
+    def __init__(self, logger, rm_single_comments, rm_block_comments):
+        self.anonDicts = {}
+        self.logger = logger
+        self.rm_single_comments = rm_single_comments
+        self.rm_block_comments = rm_block_comments
+
+    def addDicts(self, anon: str):
+        if (Regex_Dict.get(anon)):
+            self.anonDicts[anon] = Regex_Dict[anon]
+            self.logger.info("Added " + anon + "anonymizer type")
+        else:
+            self.logger.error("Processing parser: You have entered an invalid anonymizer type.") 
+
+    def anonymize(self, text: str, file: str) -> str:
+        '''
+        Removes single line and block comments and anonymises text with respect to user's options
+        '''
+        # remove single line comments such as (//, #) if user wants to
+        if self.rm_single_comments:
+            self.logger.info("[" + str(file) + "] Removing all occurances of single line comments")
+            text = re.sub(re.compile("//.*?\n" ), "", text) 
+            text = re.sub(re.compile("#.*?\n" ), "", text)
+
+        # remove block line comments such as (/*COMMENT */, ''' COMMENT ''') if user wants to
+        if self.rm_block_comments:
+            self.logger.info("[" + str(file) + "] Removing all occurances of block line comments")
+            text = re.sub(re.compile("/\*.*?\*/",re.DOTALL ) ,"" ,text) 
+            text = re.sub(re.compile("'''.*?\n" ), "", text)
+
+        # anonymises specfic anonymiser types that user wants
+        for key in self.anonDicts:
+            self.logger.info("[" + str(file) + "] Starting to anonymize " + key + " anoymizer type")
+            text = self.regex_anonymizer(text, self.anonDicts.get(key), key, file)
+        return text
+
+    def getHash(self, word):
+        hash_object = hashlib.sha256(word.encode('utf-8'))
```

Comment:
Need salt to prevent very easy re-identification and replicability?

Reply:
Have changed this accordingly to add salt.